packages/core/src/odata/common/uri-conversion/get-filter.ts

```diff
@@ -130,6 +134,42 @@ export function createGetFilter(uriConverter: UriConverter) {
         uriConverter.convertToUriFormat(filter.value, filter.edmType!)
       ].join(' ');
     }
+
+    if (isFilterLambdaExpression(filter)) {
+      return getODataFilterExpressionWhenBeingFilterLambdaExpression(
+        filter,
+        parentFieldNames,
+        targetEntityConstructor
+      );
+    }
+  }
+
+  function getODataFilterExpressionWhenBeingFilterLambdaExpression<
+    FilterEntityT extends EntityBase
+  >(
+    filter: FilterLambdaExpression<FieldType>,
+    parentFieldNames: string[] = [],
+    targetEntityConstructor: Constructable<any>
+  ) {
+    const alias = 'a';
+    if (typeof filter.innerFilter.field !== 'string') {
```

Comment:
Is this a general limitation if the `any` and `all` filters that only string is allowed or just a limitation form our side for a first implementation? If the latter is true add a  TODO and a potential follow up ticket.

Reply:
The other case would be a filter function. But I would recommend to actually cover this case by reusing the existing `getODataFilterExpression` method recursively instead of duplicating the code.

Reply:
I tested the filter function in lambda expressions with the reference services, I got 400.
I guess supporting all the types in the Filterable is easier, so you just call `getODataFilterExpression ` recursively.
Discussed with Marika, the filter function will not be supported, if there is no evidence that it should work.